include/aspect/material_model/interface.h

```diff
@@ -359,9 +359,10 @@ namespace aspect
       template <int dim>
       void average (const AveragingOperation operation,
                     const typename DoFHandler<dim>::active_cell_iterator &cell,
-                    const Quadrature<dim>   &quadrature_formula,
-                    const Mapping<dim>      &mapping,
-                    MaterialModelOutputs    &values);
+                    const Quadrature<dim>     &quadrature_formula,
+                    const Mapping<dim>        &mapping,
+                    MaterialModelOutputs      &values_out,
+                    const MaterialModelInputs<dim>  &values_in);
     }
```

Comment:
On 05/24/2015 06:48 PM, MFraters wrote:
> Ok, moved it to the end, because it was originally at the end. Or do you have
> a preference for putting both the input and output at the beginning?

The general rule is: All input arguments before all output arguments. Within 
each of these groups, list them in an order so that they make a logical 
hierarchy. E.g., if you have a DoFHandler and a solution vector, list the 
DoFHandler first and the solution vector second, since the second defines a 
field on the first.